src/macos.rs

```diff
@@ -89,49 +72,21 @@ impl TrashContext {
 
 fn delete_using_file_mgr(full_paths: Vec<String>) -> Result<(), Error> {
     trace!("Starting delete_using_file_mgr");
-    let url_cls = class!(NSURL);
-    let file_mgr_cls = class!(NSFileManager);
-    let file_mgr: id = unsafe { msg_send![file_mgr_cls, defaultManager] };
+    let file_mgr = unsafe { NSFileManager::defaultManager() };
     for path in full_paths {
-        let string = to_ns_string(&path);
+        let string = NSString::from_str(&path);
+
         trace!("Starting fileURLWithPath");
-        let url: id = unsafe { msg_send![url_cls, fileURLWithPath:string.ptr] };
-        if url == nil {
-            return Err(Error::Unknown {
-                description: format!("Failed to convert a path to an NSURL. Path: '{path}'"),
-            });
-        }
+        let url = unsafe { NSURL::fileURLWithPath(&string) };
         trace!("Finished fileURLWithPath");
-        // WARNING: I don't know why but if we try to call release on the url, it sometimes
-        // crashes with SIGSEGV, so we instead don't try to release the url
-        // let url = OwnedObject { ptr: url };
```

Comment:
The reason is that `fileURLWithPath:` returns an autoreleased instance, as documented in the [memory management rules](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmRules.html). `objc2` handles these details for you.